armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -201,13 +282,44 @@ def _getLinkedComponents(self, b, c):
 
         self.linkedComponents[c] = lstLinkedC
 
-        if lstLinkedC[0] is None:
+        if self.linkedBlocks[b].lower is None and lstLinkedC.lower is None:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked below it!",
                 single=True,
             )
-        if lstLinkedC[1] is None:
+        if self.linkedBlocks[b].upper is None and lstLinkedC.upper is None:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked above it!",
                 single=True,
             )
+
+    @staticmethod
+    def areAxiallyLinked(componentA: Component, componentB: Component) -> bool:
+        """Check if two components are axially linked.
```

Comment:
@keckler This change here means it's possible to have a subclass of the `AssemblyAxialLinkage` provide its own way to check if components should be linked. It's not _super_ easy, but possible. Rough steps:

1. Application brings a custom axial expansion changer subclass via hooks - #1870 
2. Special axial expansion changer subclass uses a subclass of `AssemblyAxialLinkage`
3. Special `AssemblyAxialLinkage` overrides this method and adds custom logic as necessary.

It would work in a pinch to help with https://github.com/terrapower/armi/issues/769